@utahdts/utah-design-system/react/components/containers/tabs/TabGroup.jsx

```diff
@@ -38,12 +50,39 @@ export function TabGroup({
           setTabGroupState((draftState) => { draftState.selectedTabId = tabId; });
         }
       },
+      navigateNext: () => {
+        const index = findCurrentTabIndex();
+        if (index !== (tabGroupState.tabs.length - 1)) {
+          navigateTab(tabGroupState?.tabs?.[index + 1] || null);
+        } else {
+          navigateTab(tabGroupState?.tabs?.[0] || null);
+        }
+      },
+      navigatePrevious: () => {
+        const index = findCurrentTabIndex();
+        if (index !== 0) {
+          navigateTab(tabGroupState?.tabs?.[index - 1] || null);
+        } else {
+          navigateTab(tabGroupState?.tabs?.[tabGroupState.tabs.length - 1] || null);
+        }
+      },
+      isVertical: !!isVertical,
     }),
     [onChange, setTabGroupState, tabGroupState, value]
   );
+
+  useEffect(() => {
+    if (tabGroupRef?.current) {
+      setTabGroupState((draftState) => {
+        // @ts-ignore
+        draftState.tabs = [...tabGroupRef?.current?.querySelectorAll('button[role=tab]') || []];
+      });
+    }
+  }, []);
+
   return (
     <TabGroupContext.Provider value={contextValue}>
-      <div className={joinClassNames('tab-group', className)} id={`tab-group-${tabGroupState.tabGroupId}`}>
+      <div className={joinClassNames('tab-group', className, isVertical && 'tab-group--vertical')} id={`tab-group-${tabGroupState.tabGroupId}`} ref={tabGroupRef}>
```

Comment:
line is getting a little long and hard to see all the props? maybe break it out to separate lines for each prop?

```
<div
className={...}
id={...}
ref={...}
>
```